client/src/main/com/sinch/sdk/models/E164PhoneNumber.java

```diff
@@ -1,11 +1,20 @@
 package com.sinch.sdk.models;
 
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class E164PhoneNumber {
+
+  private static final Pattern pattern = Pattern.compile("^\\+(?:[0-9] ?){6,14}[0-9]$");
```

Comment:
The naming convention for a private static final object would be `PATTERN`. May you can rename it to E164_PATTERN.
Also, I'm not sure the regexp is valid as a country code cannot start with 0.
Moreover, the pattern would be different depending on the country and even if a number is matching the pattern, it can still be invalid, such as +330000000000

Reply:
`pattern` -> `PATTERN` : 👍 
country code and zero: will investigate
About pattern different according to country, I was not thinking going to the details level. Aim was just about E164 format validation but not (yet) validation by country

Reply:
Updated:
- pattern -> PATTERN
- take into account invalid region code starting with zero
- by choice: do not invalid number due to specific region/country. It could be a further enhancement

Because of server is accepting strings like `+330123456..` but also `+33 0123456..`, `+(33) 0123456..` these cases are also valid cases from client side

Reply:
On what source your pattern is based?